dss/api/search.py

```diff
@@ -1,42 +1,111 @@
+    # {
+    #   "sort": [
+    #     "_doc"
+    #   ]
+    # }
+    sort = {"sort": ["_doc"]}
+
+    # TODO: (tsmith12) determine if a search operation timeout limit is needed
     try:
         es_client = ElasticsearchClient.get(get_logger())
-        search_obj = Search(using=es_client,
-                            index=Config.get_es_index_name(ESIndexType.docs, Replica[replica]),
-                            doc_type=ESDocType.doc.name).update_from_dict(es_query)
+        if _scroll_id is None:
+            page = es_client.search(index=Config.get_es_index_name(ESIndexType.docs, Replica[replica]),
+                                    doc_type=ESDocType.doc.name,
+                                    scroll=scroll,
+                                    size=per_page,
+                                    body=es_query,
+                                    sort=sort
+                                    )
+            get_logger().debug("Created ES scroll instance")
+        else:
+            get_logger().debug("Retrieve ES results from scroll instance Scroll_id: %s", _scroll_id)
+            page = es_client.scroll(scroll_id=_scroll_id, scroll=scroll)
+
+        # TODO: (tsmith12) if page returns 0 hits, then all results have been found. delete search id
 
-        # TODO (mbaumann) extract version from the request path instead of hard-coding it here
-        bundles_url_base = request.host_url + 'v1/bundles/'
+        # TODO: (tsmith12) allow users to retrieve previous search results
+        _scroll_id = page['_scroll_id']
         result_list = [{
-            'bundle_id': hit.meta.id,
-            # TODO:  use dss.util.UrlBuilder for bundle_url
-            'bundle_url': bundles_url_base + hit.meta.id.replace(".", "?version=", 1) + f"&replica={replica}",
-            'search_score': hit.meta.score
-        } for hit in search_obj.scan()]
-        return jsonify({'es_query': es_query, 'results': result_list})
-
-    except ElasticsearchDslException:
-        raise DSSException(requests.codes.bad_request,
-                           "elasticsearch_query_error",
-                           "Invalid query")
-    except ElasticsearchException:
+            'bundle_id': hit['_id'],
+            'bundle_url': _build_bundle_url(hit, replica),
+            'search_score': hit['_score']
+        } for hit in page['hits']['hits']]
+
+        next_url = request.host_url + str(UrlBuilder().set(path="v1/search")
+                                          .add_query("replica", replica)
+                                          .add_query("_scroll_id", _scroll_id))
+
+        # TODO: (tsmith12) check if all results found and return request.code.ok.
+        # TODO: (tsmith12) if all results not found return request.code.partial.
+        response = make_response(jsonify({'es_query': es_query, 'results': result_list}), requests.codes.ok)
+        response.headers["next"] = next_url
```

Comment:
Dict key strings should use single quotes, according to our convention.

Reply:
will fix